src/java/org/apache/cassandra/index/sai/memory/TrieMemoryIndex.java

```diff
@@ -61,13 +64,12 @@
     private static final int MINIMUM_QUEUE_SIZE = 128;
     private static final int MAX_RECURSIVE_KEY_LENGTH = 128;
 
-
     private final MemtableTrie<PrimaryKeys> data;
     private final PrimaryKeysReducer primaryKeysReducer;
-    private final AbstractAnalyzer.AnalyzerFactory analyzerFactory;
-    private final AbstractType<?> validator;
-    private final boolean isLiteral;
-    private final Object writeLock = new Object();
+
+    private ByteBuffer minTerm;
+    private ByteBuffer maxTerm;
+    private ReentrantLock writeLock = new ReentrantLock();
```

Comment:
Add a comment describing what this lock is protecting. I guess it applies to `data`, right?